cpp/unreal_plugins/SpCore/Source/SpCore/Std.h

```diff
@@ -0,0 +1,233 @@
+        value_container_sorted_unique.erase(begin, end);
+        return value_container_sorted_unique;
+    }
+
+    template <CValueContainer TValueContainer> requires std::convertible_to<typename TValueContainer::value_type, bool>
+    static bool all(const TValueContainer& value_container)
+    {
+        return std::ranges::all_of(value_container, [](auto val) { return val; });
+    }
+
+    template <CValueContainer TValueContainer> requires std::convertible_to<typename TValueContainer::value_type, bool>
+    static bool any(const TValueContainer& value_container)
+    {
+        return std::ranges::any_of(value_container, [](auto val) { return val; });
+    }
+
+    //
+    // Key-value container (e.g., std::map) functions
+    //
+
+    template <CKeyValueContainer TKeyValueContainer>
+    static bool containsKey(const TKeyValueContainer& key_value_container, const typename TKeyValueContainer::key_type& key)
+    {
+        return key_value_container.contains(key);
+    }
+
+    template <CKeyValueContainer TKeyValueContainer>
+    static std::vector<typename TKeyValueContainer::key_type> keys(const TKeyValueContainer& key_value_container)
+    {
+        auto view = std::views::keys(key_value_container);
+        return std::vector<typename TKeyValueContainer::key_type>(view.begin(), view.end());
+    }
+
+
+    template <CKeyValueContainer TKeyValueContainer>
+    static void insert(TKeyValueContainer& key_value_container, const typename TKeyValueContainer::key_type& key, const typename TKeyValueContainer::mapped_type& value)
+    {
+        auto [itr, success] = key_value_container.insert({key, value});
+        SP_ASSERT(success);
+    }
+
+    template <CKeyValueContainer TKeyValueContainer>
+    static void insert(TKeyValueContainer& key_value_container, const typename TKeyValueContainer::key_type& key, typename TKeyValueContainer::mapped_type&& value)
+    {
+        auto [itr, success] = key_value_container.insert({key, std::forward<typename TKeyValueContainer::mapped_type>(value)});
+        SP_ASSERT(success);
+    }
+
+    template <CKeyValueContainer TKeyValueContainer>
+    static void insert(TKeyValueContainer& dest, const TKeyValueContainer& src)
+    {
+        std::vector<typename TKeyValueContainer::key_type> keys_intersection;
+        std::ranges::set_intersection(std::views::keys(dest), std::views::keys(src), std::back_inserter(keys_intersection));
+        SP_ASSERT(keys_intersection.empty());
+        dest.insert(src.begin(), src.end());
+    }
+
+    //
+    // Functions for safely reinterpreting contiguous value containers (e.g., std::initializer_list and std::vector)
+    //
+
+    template <typename TDest, CContiguousValueContainer TContiguousValueContainer>
+    static std::span<TDest> reinterpretAsSpanOf(TContiguousValueContainer& src)
+    {
+        size_t src_num_bytes = src.size() * sizeof(TContiguousValueContainer::value_type);
```

Comment:
Let's use `sizeof(typename TContiguousValueContainer::value_type)` instead of `sizeof(TContiguousValueContainer::value_type)` to be consistent with the rest of the code.

Also, isn't it necessary to use `typename` here? I wonder how it compiled without it 🤔

Reply:
Great catch. This actually only compiles on Windows! I just verified on Compiler Explorer that `typename` is required on `clang`, just as you suspected.

Reply:
Fixed.